examples/sha256.rs

```diff
@@ -0,0 +1,245 @@
+
+        array.copy_from_slice(&result[16..24]);
+        let c = u64::from_be_bytes(array);
+
+        array.copy_from_slice(&result[24..]);
+        let d = u64::from_be_bytes(array);
+
+        // println!("{} {} {} {}", a, b, c, d);
+        return s.list(&[a, b, c, d].map(|x| s.get_u64(x)));
+    }
+}
+
+impl<F: LurkField> Sha256Coprocessor<F> {
+    pub(crate) fn new(n: usize) -> Self {
+        Self {
+            n,
+            _p: Default::default(),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+enum Sha256Coproc<F: LurkField> {
+    SC(Sha256Coprocessor<F>),
+}
+
+impl<F: LurkField> Coprocessor<F> for Sha256Coproc<F> {
+    fn eval_arity(&self) -> usize {
+        match self {
+            Self::SC(c) => c.eval_arity(),
+        }
+    }
+
+    fn simple_evaluate(&self, s: &mut Store<F>, args: &[Ptr<F>]) -> Ptr<F> {
+        match self {
+            Self::SC(c) => c.simple_evaluate(s, args),
+        }
+    }
+}
+
+impl<F: LurkField> CoCircuit<F> for Sha256Coproc<F> {
+    fn arity(&self) -> usize {
+        match self {
+            Self::SC(c) => c.arity(),
+        }
+    }
+    fn synthesize<CS: ConstraintSystem<F>>(
+            &self,
+            cs: &mut CS,
+            g: &GlobalAllocations<F>,
+            store: &Store<F>,
+            input_exprs: &[AllocatedPtr<F>],
+            input_env: &AllocatedPtr<F>,
+            input_cont: &AllocatedContPtr<F>,
+        ) -> Result<(AllocatedPtr<F>, AllocatedPtr<F>, AllocatedContPtr<F>), SynthesisError> {
+        match self {
+            Self::SC(c) => c.synthesize(cs, g, store, input_exprs, input_env, input_cont)
+        }
+    }
+}
+
+// cargo run --example sha256 1 f5a5fd42d16a20302798ef6ed309979b43003d2320d9f0e8ea9831a92759fb4b false
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    let num_of_64_bytes = args[1].parse::<usize>().unwrap();
+    let expect = hex::decode(args[2].parse::<String>().unwrap()).unwrap();
+    let _setup_only = args[3].parse::<bool>().unwrap();
+
+    let input_size = 64 * num_of_64_bytes;
+    let _input_str = vec![0u8; input_size];
+
+    let store = &mut Store::<Fr>::new();
+    let mut lang = Lang::<Fr, Sha256Coproc<Fr>>::new();
+    let sym_str = format!(".sha256.hash-{}-zero-bytes", input_size).to_string();
+    let name = Sym::new(sym_str.clone());
+    let coprocessor: Sha256Coprocessor<Fr> = Sha256Coprocessor::<Fr>::new(input_size);
+    let coproc = Sha256Coproc::SC(coprocessor);
+
+    lang.add_coprocessor(name, coproc, store);
+
+    let coproc_expr = format!("({})", sym_str);
+
+    let mut u: [u64;4] = [0u64; 4];
+    
+    for i in 0..4 {
+        u[i] = u64::from_be_bytes(expect[(i * 8) .. (i + 1) * 8].try_into().unwrap())
+    }
+    let result_expr = format!("({}u64 {}u64 {}u64 {}u64)", u[0], u[1], u[2], u[3]);
+
+    let expr = format!("(emit (eq {coproc_expr} (quote {result_expr})))");
+    let ptr = store.read(&expr).unwrap();
+    
+    let reduction_count = 1;
```

Comment:
it's worth replacing this local parameter with a CONSTANT defined at the top of the file.